app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -263,7 +271,11 @@ private void setupUI(boolean isOrientationChange) {
             gridView.setVisibility(View.GONE);
             EntitySelectViewSetup.setupDivider(this, listView, shortSelect.usesEntityTileView());
         }
-        visibleView.setOnItemClickListener(this);
+        RxAdapterView.itemClickEvents(visibleView)
+                .subscribeOn(AndroidSchedulers.mainThread())
+                .throttleFirst(CLICK_DEBOUNCE_TIME, TimeUnit.MILLISECONDS)
+                .subscribe((Consumer<AdapterViewItemClickEvent>)clickEvent ->

```

Comment:
Hey, should we unsubscribe from these events when activity gets killed ?

Reply:
If I understand library author's explanation from this link https://github.com/JakeWharton/RxBinding/issues/282
we not need to unsubscribe, because in our case we do subscribe in activity where visibleView is part of Activity view. 
Correct me if I get this explanation wrong? 
Then again we could unsubscribe.
@shubham1g5 if you think we should unsubscribe, I will do make it.

Reply:
Hey sounds right! Thanks for looking into it.